### PR TITLE
GF-21 - Network req to get yesterdays most invested project

### PR DESCRIPTION
**Pull request by iamdavidpetric**

### CODE REVIEW

**Review by MIKALAKE — APPROVED** [review]

File: src/process/routes/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import * as Paths from './paths';
+
+const Home = {
+  element: Pages.Home,
+  path: Paths.HOME_PATH
+};
+
+const routes = [Home];
+
+export default routes;
```
Comment (iamalextodoran) [inline src/process/routes/index.js]: I don't see you using this routes file. Either delete it, either use it properly.

File: src/process/routes/paths.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const HOME_PATH = '/';
```
Comment (iamalextodoran) [inline src/process/routes/paths.js:1]: Don't you also have a `project` page?

File: src/show/Application.js [hunk 3]
```diff
@@ -6,8 +6,8 @@ import { Navbar } from '../show/components';
 const Application = () => {
   return (
     <div className='flex flex-col justify-between min-h-screen h-screen w-screen'>
-      <Navbar />
       <BrowserRouter>
```
Comment (iamalextodoran) [inline src/show/Application.js]: Move `BrowserRouter` to be the first child in this return.

File: src/show/components/Button/index.js [hunk 4]
```diff
@@ -20,7 +21,8 @@ const Button = ({
   return (
     <button
       className={`flex px-2 transition-all duration-300 ${buttonColor[variant]} ${className}`}
```
Comment (iamalextodoran) [inline src/show/components/Button/index.js:25]: Does this button actually look like a disabled button if i set this to disabled={true}?

File: src/show/components/Card/index.js [hunk 5]
```diff
@@ -4,26 +4,26 @@ import { Button, ProgressBar } from '../../components';
...
-    <div className='flex flex-col w-full h-32 rounded-lg ml-2 mr-5'>
+    <div className='flex flex-col w-full h-32 rounded-lg mx-2'>
       <div className={`flex flex-row w-full h-full ${ifReversed}`}>
-        <div className='w-full rounded-lg'>
+        <div className={`w-full rounded-lg ${reversedMargin}`}>
           <img
             src={src}
             alt='Backpack'
-            className={`w-full h-full object-center object-cover rounded-lg ${ifImageMargin}`}
+            className={`w-full h-full object-center object-cover rounded-lg`}
```
Comment (iamalextodoran) [inline src/show/components/Card/index.js]: Backticks and curly braces, yet no need for them.

File: src/show/components/Card/index.js [hunk 6]
```diff
@@ -4,26 +4,26 @@ import { Button, ProgressBar } from '../../components';
...
 
   return (
-    <div className='flex flex-col w-full h-32 rounded-lg ml-2 mr-5'>
+    <div className='flex flex-col w-full h-32 rounded-lg mx-2'>
       <div className={`flex flex-row w-full h-full ${ifReversed}`}>
-        <div className='w-full rounded-lg'>
+        <div className={`w-full rounded-lg ${reversedMargin}`}>
           <img
             src={src}
             alt='Backpack'
```
Comment (iamalextodoran) [inline src/show/components/Card/index.js]: Always a backpack?

File: src/show/components/Navbar/index.js [hunk 7]
```diff
@@ -5,9 +5,13 @@ import { GiTreeGrowth } from 'react-icons/gi';
 import { MdOutlineExplore } from 'react-icons/md';
 import { IoConstructOutline } from 'react-icons/io5';
 
+import { HOME_PATH } from '../../../process/routes/paths';
+import { useNavigate } from 'react-router-dom';
 import { Button } from '../';
```
Comment (iamalextodoran) [inline src/show/components/Navbar/index.js]: Group and sorted those properly.

File: src/show/pages/public/Home/index.js [hunk 8]
```diff
@@ -1,6 +1,26 @@
...
+      .then(res => setFeaturedProject(res.data))
+      .catch(err => err);
+  }, []);
+
+  useEffect(() => {
+    axios
+      .get(`http://localhost:3000/v1/todays_project/`)
+      .then(res => setMostInvested(res.data))
+      .catch(err => err);
+  }, []);
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.js]: You have the same array of dependencies for both of your useEffects. Use only one, so fetch data with axios in only one of them.

File: src/show/pages/public/Home/index.js [hunk 9]
```diff
@@ -22,59 +42,46 @@ const Home = () => {
...
-              src='https://i.imgur.com/kHgmAGw.jpg'
-              alt='backpack'
-              className='w-full h-full object-cover rounded-t-lg'
-            />
+            {featuredProject?.images &&
+              featuredProject?.images.length &&
+              featuredProject?.images[0] && (
+                <img
+                  src={featuredProject?.images[0]}
+                  alt='backpack'
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.js]: Always a backpack?

File: src/show/pages/public/Home/index.js [hunk 10]
```diff
@@ -22,59 +42,46 @@ const Home = () => {
...
-                reversed={true}
-                title='All Access | Outdoor Leisure Backpac'
-                price='49$'
-                percentage={89}
-              />
-            </div>
-            <div className='mt-2 mr-2'>
+          {mostInvested.map((project, index) => (
+            <div
+              className={`mb-2 flex flex-col content-start ${''} `}
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.js]: Back ticks and curly braces yet no need for them.

File: src/show/pages/public/Home/index.js [hunk 11]
```diff
@@ -22,59 +42,46 @@ const Home = () => {
...
+              key={index}>
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.js]: You can now use the project's id as your key.

File: src/show/pages/public/Home/index.js [hunk 12]
```diff
@@ -22,59 +42,46 @@ const Home = () => {
...
               <Card
+                reversed={index > 1}
                 src='https://i.imgur.com/O4vo5Uf.jpg'
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.js]: What's this src?

**Review by iamalextodoran — CHANGES_REQUESTED** [review]
PR title not correct.

File: src/process/routes/index.js (hunk 1 above)
Comment (iamdavidpetric) [reply src/process/routes/index.js]: okay boso

File: src/show/Application.js (hunk 3 above)
Comment (iamdavidpetric) [reply src/show/Application.js]: ok

File: src/show/Application.js [hunk 13]
```diff
@@ -1,19 +1,24 @@
-import { BrowserRouter, Routes, Route } from 'react-router-dom';
+import routes from '../process/routes/index';
 
-import { Home, Project } from './pages';
 import { Navbar } from '../show/components';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
```
Comment (iamalextodoran) [inline src/show/Application.js]: Bobitz, what is dis?

File: src/show/components/Button/index.js [hunk 14]
```diff
@@ -2,25 +2,29 @@ import PropTypes from 'prop-types';
...
   variant
 }) => {
   const buttonColor = {
-    primary: 'text-primary hover:text-primary-700',
-    secondary: 'text-white hover:text-secondary-200',
-    outline:
-      'border-2 rounded-full border-primary-50 text-primary-50 hover:border-white hover:bg-primary-50 hover:text-white',
-    'invert-outline':
-      'border-2 rounded-full border-green-500 text-greeen-500 border-white bg-green-500 py-1 text-white hover:bg-green-600'
+    primary: disabled ? `text-primar` : `text-primary hover:text-primary-700 `,
```
Comment (iamalextodoran) [inline src/show/components/Button/index.js]:
```suggestion
    primary: disabled ? `text-primary` : `text-primary hover:text-primary-700`,
```

**Review by MIKALAKE — APPROVED** [review]

File: src/show/components/Button/index.js [hunk 15]
```diff
@@ -2,25 +2,29 @@ import PropTypes from 'prop-types';
...
+    secondary: disabled ? `text-white` : `text-white hover:text-secondary-200`,
+    outline: disabled
+      ? `hidden`
```
Comment (iamalextodoran) [inline src/show/components/Button/index.js]: Style this accordingly to your button style, but no interactions.

File: src/show/pages/index.js [hunk 16]
```diff
@@ -1,4 +1,9 @@
 import Home from './public/Home';
 import Project from './public/Project';
 
-export { Home, Project };
+const object = {
+  Home,
+  Project
+};
+
+export default object;
```
Comment (iamalextodoran) [inline src/show/pages/index.js]: Why define a constant as `object`? What would be a better name @iamdavidpetric?

File: src/show/pages/public/Home/index.js [hunk 17]
```diff
@@ -1,6 +1,28 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
 import { Button, Card, ProgressBar } from '../../../components';
 
 const Home = () => {
+  const [featuredProject, setFeaturedProject] = useState({});
+  const [mostInvested, setMostInvested] = useState([]);
+  const navigate = useNavigate();
```
Comment (iamalextodoran) [inline src/show/pages/public/Home/index.js:11]: Empty line before this.

File: src/show/pages/public/Project/index.js [hunk 18]
```diff
@@ -80,17 +80,14 @@ const Project = () => {
...
-                  (`${project?.amount_invested}` / `${project?.goal}`) * `100`
-                }
-              />
+              <ProgressBar percentage={project.achieved_goal_percentage} />
             </div>
             <div className='flex items-center justify-center mt-5'>
               <Button
                 className='flex w-1/6 justify-center'
-                variant='invert-outline'
+                variant='invertoutline'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Do `inverted-outline` as your button variant, not `invertedoutline`.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/show/pages/public/Project/index.js (hunk 18 above)
Comment (iamdavidpetric) [reply src/show/pages/public/Project/index.js]: nu functioneste with -

File: src/show/Application.js [hunk 19]
```diff
@@ -1,6 +1,7 @@
 import routes from '../process/routes/index';
 
 import { Navbar } from '../show/components';
+
 import { BrowserRouter, Routes, Route } from 'react-router-dom';
```
Comment (iamalextodoran) [inline src/show/Application.js]: This goes on the first line.

File: src/show/components/Button/index.js [hunk 20]
```diff
@@ -10,10 +10,10 @@ const Button = ({
   variant
 }) => {
   const buttonColor = {
-    primary: disabled ? `text-primar` : `text-primary hover:text-primary-700 `,
+    primary: disabled ? `text-primary` : `text-primary hover:text-primary-700 `,
```
Comment (iamalextodoran) [inline src/show/components/Button/index.js]:
```suggestion
    primary: disabled ? `text-primary` : `text-primary hover:text-primary-700`,
```

File: src/show/components/Button/index.js [hunk 21]
```diff
@@ -10,10 +10,10 @@ const Button = ({
...
     secondary: disabled ? `text-white` : `text-white hover:text-secondary-200`,
     outline: disabled
-      ? `hidden`
+      ? `border-none rounded-full text-white `
```
Comment (iamalextodoran) [inline src/show/components/Button/index.js]:
```suggestion
      ? `border-none rounded-full text-white`
```

File: src/show/Application.js [hunk 22]
```diff
@@ -1,6 +1,7 @@
 import routes from '../process/routes/index';
 
 import { Navbar } from '../show/components';
```
Comment (iamalextodoran) [inline src/show/Application.js]: Group those.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

File: src/show/Application.js (hunk 19 above)
Comment (iamdavidpetric) [reply src/show/Application.js]: okay sefuu

**Review by iamalextodoran — APPROVED** [review]

**Pull request merged**